### DISCUSSION TITLE
Using CUE to send HTTP POST requests

### DISCUSSION THREAD

**Discussion by ghost** [cue-lang/cue#1664] [Q&A] [answered]
Hi there,

I am trying to utilize the CUE scripting layer to send a HTTP POST request to a webserver.
So far my script looks like this, I have encoded the payload in bytes but however, the request is not accepted.

```
command: req: {
  query: http.Do & {
    method: "POST"
    url:    "https://api.meraki.com/api/v1"
    request: {
      body: "7b226e616d653a22666f6f227d"
      header: {"Content-Type": "application/json"}
    }
  }
  print: exec.Run & {
    cmd: ["echo", query.response.body]
  }
}
```

When sending GET and DELETE requests, everything works fine. I guess it is something wrong with the encoded data.
The server is responding the following error message back: { "error": "body contains badly-formed JSON (at character 2)" }

As a second question: Is it also possible to directly encode CUE-generated values to bytes and send it within the HTTP request body?

Something like this:

```
album: {
  foo: "Led Zeppelin"
  bar: "BBC Sessions"
}
```

Encoded and then added to the HTTP request body like so:

```
// The same code from above
request: {
  body: album
}
```

Maybe some background information, what I am trying to achieve: I want to generate switch configuration for Meraki switches using 
CUE, validate the planned configuration and then send the complete payload directly as HTTP POST request to the Meraki controllers.

Thanks.

**Comment by verdverm** [comment, marked as answer]
Try using single quotes for bytes, you have a string for body

To use a CUE value, try out the encoding/json.Marshal function

**Comment by ghost** [comment]
Yeah, thanks. That was my problem. 

Using

```
data: {
  settings: {
    name: "foo"	
  }
}
```

along with

```
json.Marshal(data)
```

solved my problem.
